ONSdigital/response-operations-ui
# RAS-315 Refactor collection-exercise

# What and why?
This PR moves the logic from the frontend to the backend and puts in some best practises of type hinting and testing. There can be more done in this space, but I have had to be realistic what could be done in the time, in the end of the day we are populating a 3 column table, we should be able to do all the preparation in the backend to hand to a frontend to display


# How to test?
Elements that need to be tested.

- Edit permission - create a basic CE with an account that can, then logout and come back in with one that can't and make sure there are no links
- Check all the links are correct, i.e the text is right, edit when there is a value add when there isn't, also check they work and populate the right fields in the db
- Locking - make sure fields which should be locked, are not editable. These will be Period ID and Reporting period, or anything in the past. To lock a CE it should be in one of these states (LIVE", "READY_FOR_LIVE", "EXECUTION_STARTED", "VALIDATED", "EXECUTED", "ENDED") for this test just complete the CE and press the green button that will do it.
- Status alerts (I don't think these ever appear normal and I have no idea why they are here, but I have migrated them) The only way you can do this is go to the db and change the status of the event (SELECT * FROM collectionexercise.event) Change to FAILED or RETRY etc.
- Make sure collection instruments and sample work the same
- Make sure reminders/nudges generated as before

# Trello
